aws/src/test/java/org/apache/iceberg/aws/TestAwsProperties.java

```diff
@@ -204,4 +206,21 @@ public void testS3FileIoSessionCredentialsConfiguration() {
         "secret",
         capturedAwsCredentialsProvider.resolveCredentials().secretAccessKey());
   }
+
+  @Test
+  public void testKryoSerialization() throws IOException {
```

Comment:
Can you also add a test for S3FileIO or whatever was causing this to get serialized? I think we must be missing a test for that as well.

Reply:
The no arg constructor of AwsProperties is not in common code path, and it is overwritten in `S3FileIO.initialize()`, so usually it's not actually used if `S3FileIO.initialize()` is called.
I think AwsProperties's no arg constructor is no longer needed? Or we can keep it and make it call the other constructor:
`public AwsProperties() {
    this(Collections.emptyMap());
  }`

Reply:
@rdblue added a test for S3FileIO, also:
-  refactored constructor of `AwsProperties` to remove dup code
- applied a similar fix to `PropertyUtil` that is used in `AwsProperties`